sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/RenewTokenWorker.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.android.communication.chat.implementation.notifications.fcm;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.work.Data;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+import com.azure.android.communication.chat.implementation.notifications.NotificationUtils;
+import com.azure.android.communication.common.CommunicationTokenCredential;
+
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Worker for renewing registra token. This worker could execute when APP closed. This is introduced to allow renewal
+ * happen even when APP gets closed.
+ * https://skype.visualstudio.com/SPOOL/_queries/query/af989357-74be-49b6-833a-d4359cd1cbd4/
+ */
+public class RenewTokenWorker extends Worker {
+    private RegistrarClient registrarClient;
+
+    private CommunicationTokenCredential communicationTokenCredential;
+
+    private RenewalWorkerDataContainer renewalWorkerDataContainer;
+
+    public RenewTokenWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
+        super(context, workerParams);
+    }
+
+    public RenewTokenWorker(@NonNull Context context, @NonNull WorkerParameters workerParams,
+                            CommunicationTokenCredential communicationTokenCredential) {
+        super(context, workerParams);
+        this.registrarClient = new RegistrarClient();
+        this.communicationTokenCredential = communicationTokenCredential;
+        this.renewalWorkerDataContainer = RenewalWorkerDataContainer.instance();
+    }
+
+    @NonNull
+    @Override
+    public Result doWork() {
+        int attempts = getRunAttemptCount();
+        Log.i("RenewTokenWorker", "RenewTokenWorker execution in background: " + attempts);
```

Comment:
We should use `ClientLogger` to log accross our libraries, it uses the Android logger by default.